control/timeresp.py

```diff
@@ -826,60 +830,187 @@ def impulse_response(sys, T=None, X0=0., input=0, output=None, T_num=None,
     return T, yout
 
 # utility function to find time period and time increment using pole locations
-def _ideal_tfinal_and_dt(sys):
-    constant = 7.0
-    tolerance = 1e-10
-    A = ssdata(sys)[0]
-    if A.shape == (0,0): 
-        # no dynamics
-        tfinal = constant * 1.0 
-        dt = sys.dt if isdtime(sys, strict=True) else 1.0
-    else:
-        poles = sp.linalg.eigvals(A)
-        if isdtime(sys, strict=True):
-            poles = np.log(poles)/sys.dt # z-poles to s-plane using s=(lnz)/dt
-        
-        # calculate ideal dt
-        if isdtime(sys, strict=True):
-            dt = sys.dt
-        else:
-            fastest_natural_frequency = max(abs(poles))
-            dt = 1/constant / fastest_natural_frequency
+def _ideal_tfinal_and_dt(sys, is_step=True):
+    """helper function to compute ideal simulation duration tfinal and dt, the 
+    time increment. Usually called by _default_time_vector, whose job it is to 
+    choose a realistic time vector. Considers both poles and zeros.  
+    
+    For discrete-time models, dt is inherent and only tfinal is computed.
+
+    Parameters
+    ----------
+    sys : StateSpace or TransferFunction
+        The system whose time response is to be computed
+    is_step : bool
+        Scales the dc value by the magnitude of the nonzero mode since
+        integrating the impulse response gives ∫exp(-λt) = -exp(-λt)/λ.
+        Default is True.
         
-        # calculate ideal tfinal
-        poles = poles[abs(poles.real) > tolerance] # ignore poles near im axis
-        if poles.size == 0:        
-            slowest_decay_rate = 1.0
-        else:
-            slowest_decay_rate = min(abs(poles.real))
-        tfinal = constant / slowest_decay_rate  
+    Returns
+    -------
+    tfinal : float
+        The final time instance for which the simulation will be performed.
+    dt : float
+        The estimated sampling period for the simulation.
+    
+    Notes
```

Comment:
I think this part can go into code comments instead. Nobody needs to read a wall of text probably. But just my opinion, feel free to dismiss.

Reply:
I think it doesn't hurt. That's what the Notes section is for.